include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -233,6 +233,14 @@ struct __atomic_ref : sycl::atomic<_AtomicType, _Space>
 };
 #endif // _ONEDPL_SYCL2023_ATOMIC_REF_PRESENT
 
+template <typename DataT, sycl::access::mode AccessMode, int Dimensions = 1>
```

Comment:
```sycl::access::mode AccessMode``` template param required here only for compatibility with old DPC++ compilers.

Reply:
I would remove that template parameter here... According to new local accessor API, writing mode is defined by "non-const" DataT - see `4.7.6.11.2. Read only local accessors and implicit conversions` from the spec:
`Since local_accessor has no template parameter for the access mode, the only specialization for a read-only local accessor is by providing a const qualified DataT parameter. Specializations with a non-const qualified DataT parameter are read-write.`

So, in the deprecated code 
`sycl::accessor<DataT, Dimensions, AccessMode....`
to "inference" type of AccessMode you may use `std::is_const`

Reply:
In our old code we have next variants:
- access_mode::read_write
- access_mode::discard_read_write

What do you propose to do instead of this old template param?

Reply:
according to SYCL spec (old versions), semantic of `access_mode::discard_read_write` is the same as `access_mode::discard_write`... but any case, for the shared local memory "discard_write" doesnt make sense..
Moreover, just "reading" looks weird for the shared local memory...because I there is no local accessor constructor with data initialization... So, no way to initialize slm data via SYCL API for further just reading...  Probably, just using low level API or getting pointer and casting to non-const pointer allows to write something to that slm for further reading, But it is also weird because creating local_accessor with reda_write mode (non-const DataT) is better and more correct way.

Reply:
Possible you are probably right.
But I think that at this pull request we should just stop using the deprecated class and not introduce other changes, otherwise the risk of errors increases.

Reply:
BTW, also SYCL spec tells 'There is an implicit conversion from the read-write specialization to the read-only specialization, provided that all other template parameters are the same."
So, local_accessor::swap is usefull here.

Reply:
No risk at all, the changes are trivial.

Reply:
Fixed